core/include/moveit/task_constructor/task.h

```diff
@@ -47,6 +47,26 @@
 #include <moveit/macros/class_forward.h>
 
 #include <moveit_msgs/MoveItErrorCodes.h>
+#include <moveit/task_constructor/moveit_compat.h>
+
+#if MOVEIT_CORE_HAS_ERROR_CODE
+#include <moveit/utils/moveit_error_code.h>
+#else
+namespace moveit {
+namespace core {
+class MoveItErrorCode : public moveit_msgs::MoveItErrorCodes
+{
+public:
+	MoveItErrorCode() { val = 0; }
+	MoveItErrorCode(int code) { val = code; }
+	MoveItErrorCode(const moveit_msgs::MoveItErrorCodes& code) { val = code.val; }
+	explicit operator bool() const { return val == moveit_msgs::MoveItErrorCodes::SUCCESS; }
+	bool operator==(const int c) const { return val == c; }
+	bool operator!=(const int c) const { return val != c; }
+};
+}  // namespace core
+}  // namespace moveit
```

Comment:
I'm hesitant to include a redefinition of MoveItErrorCode. I would prefer to backport: https://github.com/ros-planning/moveit/pull/3015
If that's not desired, just reuse an existing definition via:
`using MoveItErrorCode = moveit::planning_interface::MoveItErrorCode;`